src/views/communities/index.tsx

```diff
@@ -181,37 +181,45 @@ function CommunitiesScreen() {
         );
     };
 
+    const ITEM_HEIGHT = 147;
+
+    const getItemLayout = useCallback(
+        (data, index) => ({
+            length: ITEM_HEIGHT,
+            offset: ITEM_HEIGHT * index,
+            index,
+        }),
+        []
+    );
+
+    const renderItem = useCallback(
+        ({ item, index }: { item: CommunityAttributes; index: number }) =>
+            index === 0 ? filterHeader() : <CommunityCard community={item} />,
+        []
+    );
+
+    const keyExtractor = useCallback((item) => item.publicId, []);
+
     return (
         <>
             <FlatList
+                ref={flatListRef}
                 data={[
                     { publicId: 'for-compliance-sake-really' } as any,
                 ].concat(communities)}
-                renderItem={({
-                    item,
-                    index,
-                }: {
-                    item: CommunityAttributes;
-                    index: number;
-                }) =>
-                    index === 0 ? (
-                        filterHeader()
-                    ) : (
-                        <CommunityCard community={item} />
-                    )
-                }
-                ref={flatListRef}
-                keyExtractor={(item) => item.publicId}
+                keyExtractor={keyExtractor}
+                renderItem={renderItem}
                 onEndReachedThreshold={0.7}
                 onEndReached={handleOnEndReached}
                 showsVerticalScrollIndicator={false}
                 // Performance settings
                 removeClippedSubviews // Unmount components when outside of window
-                initialNumToRender={2} // Reduce initial render amount
-                maxToRenderPerBatch={1} // Reduce number in each render batch
+                initialNumToRender={4} // Reduce initial render amount
```

Comment:
why increasing?

Reply:
Again, we have a tradeoff here. Our screen (most of time) shows 2 communities per screen and we're loading just that .After the first scroll we had another call to render more which was increasing the "freeze" perception. These changes are based on (https://reactnative.dev/docs/optimizing-flatlist-configuration). The value of 2 is "just enough" but in my opinion, too tight.

Reply:
Nice. Agree with that :+1: